src/Symfony/Component/PropertyAccess/Exception/UnexpectedPropertyTypeException.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\PropertyAccess\Exception;
+
+use Symfony\Component\PropertyAccess\PropertyPathInterface;
+
+/**
+ * Thrown when a property value within the path is neither object nor array.
+ *
+ * @author Mikael Pajunen <mikael.pajunen@gmail.com>
+ */
+class UnexpectedPropertyTypeException extends UnexpectedTypeException
+{
+    public function __construct($value, PropertyPathInterface $path, $index)
+    {
+        // Skip parent constructor.
+        RuntimeException::__construct(sprintf(
```

Comment:
It isnt a hack. https://stackoverflow.com/questions/12589289/how-to-skip-execution-of-a-parent-method-to-execute-grandparent-method